templates/certification/hardware-details.html

```diff
@@ -1,35 +1,94 @@
-<h1>This is the hardware details page</h1>
+{% extends "certification/base_certification.html" %}
 
-<h2>ID (Canonical ID)</h2>
-{{canonical_id}}
+{% block title %} {{ vendor }} {{ model }} certified with Ubuntu {{ release }} | Ubuntu{% endblock %}
 
-<h2>Name:</h2>
-{{ name }}
+{% block meta_description %}Hardware that have been certified for use with Ubuntu.{% endblock meta_description %}
+{% block meta_copydoc %}https://docs.google.com/document/d/1QuhO-9FEOGLrYp8bErS_9snqdljl7d6tFAUoNQxoVDQ/edit{% endblock meta_copydoc %}
 
-<h2>Category/Form</h2>
-{{ category }}
+{% block content %}
 
-<h2>form_factor</h2>
-{{ form_factor }}
+<section class="p-strip--suru-topped">
+  <div class="row u-sv3">
+    <div class="col-7">
+      <p class="p-heading--4">
+        <a href="/certification/{{ release_details.canonical_id }}">{{ release_details.make }} {{ release_details.model }} certified on Ubuntu {{ release_details.certified_release }}</a>
+      </p>
+      <h1 class="p-heading--2">Hardware details</h1>
+    </div>
+    <div class="col-5 u-align--center u-vertically-center u-hide--small">
+    {% if release_details.category == "Desktop" or release_details.category == "Laptop" %}
+      {{ image (
+        url="https://assets.ubuntu.com/v1/4b732966-Laptop.svg",
+        alt="",
+        width="132",
+        height="77",
+        hi_def=True,
+        loading="auto"
+        ) | safe
+      }}
+      {% endif %}
+      {% if release_details.category == "Server" %}
+      {{ image (
+        url="https://assets.ubuntu.com/v1/fdf83d49-Server.svg",
+        alt="Server",
+        width="80",
+        height="96",
+        hi_def=True,
+        loading="auto"
+        ) | safe
+      }}
+      {% endif %}
+      {% if release_details.category == "Ubuntu Core" %}
+      {{ image (
+        url="https://assets.ubuntu.com/v1/f9f8ace9-gateway.svg",
+        alt="Gateway",
+        width="96",
+        height="100",
+        hi_def=True,
+        loading="auto"
+        ) | safe
+      }}
+      {% endif %}
+      {% if release_details.category == "Server SoC" %}
+      {{ image (
+        url="https://assets.ubuntu.com/v1/4e0399a1-chip.svg",
+        alt="Chip",
+        width="84",
+        height="100",
+        hi_def=True,
+        loading="auto"
+        ) | safe
+      }}
+      {% endif %}
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-12">
+      <table class="p-table-hardware-details" aria-label="Hardware details for {{ release_details.make }} {{ release_details.model }}">
+        <tbody>
+          {% for category, devices in hardware_details.items() %}
+            {% if category != "Other" %}
+            <tr>
+              <th class="p-muted-text">
+                {{ category }}
+              </th>
+              <td>
+                <ul class="p-list u-no-margin--bottom">
+                {% for device in devices %}
+                <p>{{ device.id }}</p>
```

Comment:
Why do you need `<p>` inside of lists? can you not use `<li>`?
According to [W3C](https://validator.w3.org/nu/?doc=https%3A%2F%2Fubuntu-com-9598.demos.haus%2Fcertification%2F201606-22523%2F16.04%2520LTS), `<p>` are not allowed inside of lists

Reply:
Good spot!